### PR TITLE
feat (cli, api): import pipeline

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: cli/cds/pipeline/import.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	"github.com/ovh/cds/sdk"	
+)
+
+var importFormat string
+
+func importCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "cds pipeline import <projectKey> <file>",
```
Comment (yesnault) [inline cli/cds/pipeline/import.go]: Link to markdown page with doc on how write yml

Comment (yesnault) [inline cli/cds/pipeline/import.go]: --url action import action ?

File: engine/api/test/assert.go [hunk 2]
```diff
@@ -42,22 +43,48 @@ func interfaceSlice(slice interface{}) []interface{} {
 	return ret
 }
 
-func arrayContains(array interface{}, s interface{}) bool {
+func ArrayContains(array interface{}, s interface{}) bool {
```
Comment (yesnault) [inline engine/api/test/assert.go]: doc ?

File: engine/api/test/assert.go [hunk 3]
```diff
@@ -42,22 +43,48 @@ func interfaceSlice(slice interface{}) []interface{} {
...
 	for _, i := range b {
-		if reflect.DeepEqual(i, s) {
+		if DeepEquals(i, s) {
 			return true
 		}
 	}
 	return false
 }
 
+func Equal(t *testing.T, a, b interface{}, msgAndArgs ...interface{}) {
```
Comment (yesnault) [inline engine/api/test/assert.go]: doc?

File: engine/api/test/assert.go [hunk 4]
```diff
@@ -42,22 +43,48 @@ func interfaceSlice(slice interface{}) []interface{} {
...
+		t.Log("Expected:")
+		t.Log(dump.Sdump(a))
+		t.Log("Actual:")
+		t.Log(dump.Sdump(b))
+	}
+
+	assert.True(t, e, msgAndArgs...)
+}
+
+func DeepEquals(a, b interface{}) bool {
```
Comment (yesnault) [inline engine/api/test/assert.go]: doc?

File: cli/cds/pipeline/import.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			if err != nil {
+				sdk.Exit("Error: %s\n", err)
+			}
+
+			var url string
+
+			url = fmt.Sprintf("/project/%s/pipeline/import?format=%s", projectKey, importFormat)
+
+			data, code, err := sdk.Request("POST", url, btes)
+			json.Unmarshal(data, &msg)
```
Comment (sguiheux) [inline cli/cds/pipeline/import.go]: where is err?

**Comment by fsamin** [comment]
@yesnault @sguiheux doc review : https://github.com/ovh/cds/blob/fsamin/import-pipeline/doc/tutorials/pipeline-configuration-file.md

File: doc/tutorials/pipeline-configuration-file.md [hunk 6]
```diff
@@ -1,11 +1,221 @@
...
-## Format
-
 ## Pipeline Concepts
 
-## Stages
+A pipeline describes how things need to be executed in order to achieve wanted result. In CDS, a pipeline a defined on a project and can be used on several applications inside the same project.
+
+A pipeline is structured in sequential **stages** containing one or multiple concurrent **jobs**.
+
+In CDS there is several types of pipeline : **buiild**, **testing** and **deployment**. In Pipeline configuration file, default type is **build**.
```
Comment (sguiheux) [inline doc/tutorials/pipeline-configuration-file.md]: too much i on buiild

File: doc/tutorials/pipeline-configuration-file.md [hunk 7]
```diff
@@ -1,11 +1,221 @@
...
+
+A Job will be executed by a **worker**. CDS will choose and provision a worker for dependending of the **requirements** you define on your job.
+You can set as many requirements as you want, following those rules :
+
+- Only one model can be set as requirement
+- Only one hostname can be set as requirement
+- Memory and Services requirements are availabe only on Docker models
+
+If you want to share files or artifact between jobs, stages or pipelines you have to use *Artifact upload* and *Artifact download*.
+
```
Comment (sguiheux) [inline doc/tutorials/pipeline-configuration-file.md]: explain worker export command too ?

File: tests/cli_pipeline_export_import.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
+name: Pipeline Export / Import Command TestSuite
+testcases:
+- name: prepare test
+  steps:
+  - script: {{.cds}} project remove --force ITCLIPIPPRJ
```
Comment (yesnault) [inline tests/cli_pipeline_export_import.yml]: ITCLIPIPPRJ -> already exist

File: tests/cli_pipeline_export_import.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  - script: {{.cds}} group remove ITCLIPIP_GRP --force
```
Comment (yesnault) [inline tests/cli_pipeline_export_import.yml]: idem ITCLIPIP_GRP

**Pull request merged**